src/hrflow_connectors/connectors/boards/smartrecruiters/actions.py

```diff
@@ -0,0 +1,178 @@
+from pydantic import Field
+from ....core.action import BoardAction
+from ....core.http import HTTPStream
+from ....core.auth import SmartToken
+
+
+class SmartJobs(HTTPStream, BoardAction):
+    auth: SmartToken
+    updated_after: Optional[str] = Field(
+        None, description="custom pulling of jobs only updated after a certain date"
+    )
+    offset: int = Field(
+        ...,
+        description="if offset is 0 and limit is 10 and `totalFound` resources are 130, our response will contain the first 10",
+    )
+    posting_status: str = Field("PUBLIC", description="Job offers availability")
+    limit: int = Field(..., description="see `offset` description")
+
+    @property
+    def base_url(self):
+        return "https://api.smartrecruiters.com/jobs"
+
+    @property
+    def http_method(self):
+        return "GET"
+
+    def pull(self) -> Iterator[Dict[str, Any]]:
+        """
+        `pull` [send tokenized requests to SmartRecruiters to pull job offers content]
+
+        Returns:
+            [jobs_content]: Iterator[Dict[str, Any] [a list of jobs content in json format]
+        """
+        jobs_content = []
+        self.params["postingStatus"] = self.posting_status
+        self.params["limit"] = self.limit
+        self.params["offset"] = self.offset
+
+        if self.updated_after:
+            self.params["updatedAfter"] = self.updated_after
+
+        response = self.send_request()
+        total_found = response.json()["totalFound"] #total offers found
+        while self.offset < total_found:
+            self.params.update({"offset": self.offset})
+            response_update = self.send_request()
+            content = response_update.__dict__["content"]  #get list of job offers contents
+            jobs_content += content
+            self.offset += self.limit
+
+        return jobs_content
+
+    def format(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        `format`[generates a dictionary of a job attributes, for each job content data]
+
+        Args:
+            data (Dict[str, Any]): [unique job offer content data yielded after the function `pull` is executed.]
+
+        Returns:
+            Dict[str, Any]: [a job in the HrFlow job object format]
+        """
+        job = dict()
+        # job Title
+        job["name"] = data.get("title")
+        # job Reference
+        job["reference"] = data.get("refNumber")
+        # job Url
+        job["url"] = None
+        # creation date and -update- of the offer
+        job["created_at"] = data.get("createdon")
+        job["updated_at"] = data.get("updatedon")
+        job["summary"] = ""
+        # location
+        lat = data.get("location", {}).get("latitude")
+        lng = data.get("location", {}).get("longitude")
+        text = " ".join(
+            data.get("location")[key]
+            for key in ["country", "region", "city", "address"]
+            if data.get("location", {}).get(key)
+        )
```

Comment:
Don't do 10 things in 1 line of code.

You'd rather read boring code (that you know what's going to happen on the next line).
You do a `join` of something from a filtered list created on the move.

Do this :
```
location_field_name_list = ["country", "region", "city", "address"]
location_field_list = []
for field_name in location_field_list:
  ...
text = " ".join(location_field_list)
```
I give an example but it could be even more explicit!